eZ/Publish/Core/Search/Common/Slot/AbstractSubtree.php

```diff
@@ -26,24 +29,28 @@ protected function indexSubtree($locationId)
         $subtreeIds = $locationHandler->loadSubtreeIds($locationId);
 
         foreach ($subtreeIds as $locationId => $contentId) {
-            $this->searchHandler->indexLocation(
-                $locationHandler->load($locationId)
-            );
-
-            if (isset($processedContentIdSet[$contentId])) {
-                continue;
+            if ($this->searchHandler instanceof LocationIndexer) {
+                $this->searchHandler->indexLocation(
+                    $locationHandler->load($locationId)
+                );
             }
 
-            $this->searchHandler->indexContent(
-                $contentHandler->load(
-                    $contentId,
-                    $contentHandler->loadContentInfo($contentId)->currentVersionNo
-                )
-            );
-
-            // Content could be found in multiple Locations of the subtree,
-            // but we need to (re)index it only once
-            $processedContentIdSet[$contentId] = true;
+            if ($this->searchHandler instanceof ContentIndexer || $this->searchHandler instanceof FullTextIndexer) {
+                if (isset($processedContentIdSet[$contentId])) {
+                    continue;
+                }
+
+                $this->searchHandler->indexContent(
+                    $contentHandler->load(
+                        $contentId,
+                        $contentHandler->loadContentInfo($contentId)->currentVersionNo
+                    )
+                );
```

Comment:
@pspanja: I wonder if we need indexing content in case of a `FullTextIndexer` in `AbstractSubtree` at all. Slots that extend `AbstractSubtree` are not needed for the legacy search engine, I think.